platform/MCU/STM32F2xx/STM32_USB_Device_Driver/src/usbd_core.c

```diff
@@ -301,7 +301,14 @@ static uint8_t USBD_DataInStage(USB_OTG_CORE_HANDLE *pdev , uint8_t epnum)
         }
         else
         {
-          if((pdev->dev.class_cb->EP0_TxSent != NULL)&&
+          uint8_t ret = USBD_OK;
+          if (pdev->dev.usr_cb && pdev->dev.usr_cb->ControlRequest &&
+              (pdev->dev.device_status == USB_OTG_CONFIGURED))
+          {
+            ret = pdev->dev.usr_cb->ControlRequest(NULL, 1);
```

Comment:
can we make `1` a symbol/enum value? or add a comment please.

Reply:
:+1: Or change that to bool for example. It's a simple flag that indicates whether there is a data stage or not.
```
uint8_t (*ControlRequest)(USB_SETUP_REQ* req, uint8_t dataStage);
```

Reply:
I don't mind either way - whatever gives clarity when reading the code.  It's a poor consequence of reading diffs that a lot of the context is missing, but that can be a good thing since it leads us to being over-communicative in the code.